### ISSUE TITLE
IOBalancerMemoryLeakTest.testEachConnectionUseDifferentSelectorEventually

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#7222]
**maintenance**
```
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.nio.tcp.iobalancer.IOBalancerMemoryLeakTest$1.run(IOBalancerMemoryLeakTest.java:60)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:737)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:751)
	at com.hazelcast.nio.tcp.iobalancer.IOBalancerMemoryLeakTest.testEachConnectionUseDifferentSelectorEventually(IOBalancerMemoryLeakTest.java:55)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast/713/testReport/junit/com.hazelcast.nio.tcp.iobalancer/IOBalancerMemoryLeakTest/testEachConnectionUseDifferentSelectorEventually/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/817/testReport/junit/com.hazelcast.nio.tcp.nonblocking.iobalancer/IOBalancerMemoryLeakTest/testMemoryLeak/

**Comment by eminn** [comment]
possible test fix 36a39b81a7333c108a10b488417a6006427b1d6b
